setup/compiler.class.inc.php

```diff
@@ -2892,25 +2898,54 @@ protected function CompileThemes($oBrandingNode, $sTempTargetDir, $sFinalTargetD
 			// Check if a precompiled version of the theme is supplied
 			$sPostCompilationLatestPrecompiledFile = $sPostCompilationPrecompiledThemeFolder . $sThemeId . ".css";
 
-			$sPrecompiledFileToUse = $this->UseLatestPrecompiledFile($sTempTargetDir, $aThemeParameters['precompiled_stylesheet'], $sPostCompilationLatestPrecompiledFile, $sThemeId);
+			$sPrecompiledFileToUse = $this->UseLatestPrecompiledFile($sTempTargetDir, $sPrecompiledStylesheet, $sPostCompilationLatestPrecompiledFile, $sThemeId);
 			if ($sPrecompiledFileToUse != null){
 				copy($sPrecompiledFileToUse, $sThemeDir.'/main.css');
 				// Make sure that the copy of the precompiled file is older than any other files to force a validation of the signature
 				touch($sThemeDir.'/main.css', 1577836800 /* 2020-01-01 00:00:00 */);
 			}
 
-			$bHasCompiled = ThemeHandler::CompileTheme($sThemeId, true, $this->sCompilationTimeStamp, $aThemeParameters, $aImportsPaths, $sTempTargetDir);
-			if ($bHasCompiled)
-			{
-				SetupLog::Info("Replacing theme '$sThemeId' precompiled file in file $sPostCompilationLatestPrecompiledFile for next setup.");
-				copy($sThemeDir.'/main.css', $sPostCompilationLatestPrecompiledFile);
-			}else {
+			if (!static::$oThemeHandlerService) {
+				static::$oThemeHandlerService = new ThemeHandlerService();
+			}
+			$bHasCompiled = static::$oThemeHandlerService->CompileTheme($sThemeId, true, $this->sCompilationTimeStamp, $aThemeParameters, $aImportsPaths, $sTempTargetDir);
+
+			if ($bHasCompiled) {
+				if (utils::GetConfig()->Get('theme_precompilation_enabled')){
+					if (utils::IsDevelopmentEnvironment() && ! empty(trim($sPrecompiledStylesheet)))
+					{ //help developers to detect & push theme precompilation changes
+						$sInitialPrecompiledFilePath = null;
+						$aRootDirs = $this->oFactory->getRootDirs();
+						if (is_array($aRootDirs) && count($aRootDirs) !== 0) {
+							foreach ($this->oFactory->getRootDirs() as $sRootDir) {
```

Comment:
```suggestion
						$aRootDirs = $this->oFactory->GetRootDirs();
						if (is_array($aRootDirs) && count($aRootDirs) !== 0) {
							foreach ($this->oFactory->GetRootDirs() as $sRootDir) {
```

Rename to be up to date with other comment